doc/dev/dataplane_codegen_quick_start.md

```diff
@@ -0,0 +1,198 @@
+
+## Project folder structure
+
+Before we start, we probably should get to know the project folder for [SDK repo](https://github.com/Azure/azure-sdk-for-python).
+
+All the files for your SDK will reside in the **${PROJECT_ROOT} folder**, where
+
+```sh
+${PROJECT_ROOT}=sdk/{service_name}/azure-{service_name}-{module_name}
+```
+
+Normally, the folder structure would be something like:
+
+- **${PROJECT_ROOT}**
+  - `azure/{service_name}/{module_name}` : folder where generated code is.
+  - `swagger`: folder of configuration file which is used to generate SDK code with autorest
+  - `tests`: folder of test files
+  - `samples`: folder of sample files
+
+More details on the structure of Azure SDK repos is available in the [Azure SDK common repo](https://github.com/Azure/azure-sdk/blob/main/docs/policies/repostructure.md#sdk-directory-layout).
+
+## How to generate SDK code with Dataplane Codegen
+
+We are working on to automatically generate everything right now, but currently we still need some manual work to get a releasable package. Here're the steps of how to get the package.
+
+1. **Create the ${PROJECT_ROOT} folder if it does not already exist**
+
+2. **Create a README.md file under ${PROJECT_ROOT}/swagger**
+
+   This README.md file will hold the options to be passed to autorest to generate the code. Storing these options in a file in the
+   SDK repo helps make SDK generation reproducible.
+
+   ````markdown
+   # Azure Purview for Python
+  
+   > see https://aka.ms/autorest
+  
+   ### Settings
+  
+   ```yaml
+   input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json
+   output-folder: ../azure/messaging/webpubsubservice
+   namespace: azure.messaging.webpubsubservice
+   package-name: azure-messaging-webpubsubservice
+   license-header: MICROSOFT_MIT_NO_VERSION
+   clear-output-folder: true
+   no-namespace-folders: true
+   python: true
+   title: WebPubSubServiceClient
+   version-tolerant: true
+   package-version: 1.0.0b1
+   add-credential: true
+   credential-scopes: https://webpubsub.azure.com/.default
+   ```
+   ````
+
+   **Reviewers: which of the above can we remove?  We should keep only those needed for the "common" case. And if any could just be the default (e.g. license-header), we should remove those too. Then we should document whatever remains.**
+
+   There are some configuration in the template and you could reference [autorest/flags](https://github.com/Azure/autorest/blob/main/docs/generate/flags.md) to get more info. We need to replace the value in `package-name`, `title`,  `input-file`, `package-version`, `credential-scopes`, `output-folder` into **your own service's** `package-name`, `title`etc.
+
+3. **Run autorest to generate the SDK**
+
+   Now you can run this command in swagger folder you just created.
+
+   ```sh
+   autorest --version=latest --python --use=@autorest/python@latest  README.md
```

Comment:
You should not need `--version=latest` and the `--use`. We should also tell people to do a `autorest --reset`. Autorest cache is really conservative, and if you use autorest over several months, you may end up with something super old. Reset delete the cache. I reached a level of confused people, that I always tell then to do a `autorest --reset` before any commands.